test/sequential/test-repl-persistent-history.js

```diff
@@ -74,35 +74,41 @@ const oldHistoryPath = path.join(fixtures, 'old-repl-history-file.json');
 const tests = [{
   env: { NODE_REPL_HISTORY: '' },
   test: [UP],
-  expected: [prompt, replDisabled, prompt]
+  expected: [prompt, replDisabled, prompt],
+  event: 'end'
```

Comment:
The end event worked fine for me. To really see it, you can run with the `NODE_DEBUG=repl`. It makes me wonder if we should defer emitting close until after flushHistory though?